src/java.naming/share/classes/com/sun/jndi/ldap/AbstractLdapNamingEnumeration.java

```diff
@@ -42,17 +45,48 @@
 
     protected Name listArg;
 
-    private boolean cleaned = false;
-    private LdapResult res;
-    private LdapClient enumClnt;
     private Continuation cont;  // used to fill in exceptions
     private Vector<LdapEntry> entries = null;
     private int limit = 0;
     private int posn = 0;
-    protected LdapCtx homeCtx;
+
     private LdapReferralException refEx = null;
     private NamingException errEx = null;
 
+    /* This class maintains the pieces of state that need (or are needed for)
+     * cleanup, which happens by calling cleanup(), or is done by the Cleaner.
+     */
+    private static class CleaningAction implements Runnable {
```

Comment:
This nested class might be more aptly named  `EnumCtx` since it is used during the enumeration process.

The mention of the `cleanup()` method in the nested class javadoc is a bit ambiguous, it seems there should be a cleanup() method in the class, but it is in AbstractLdapNamingEnumeration.

The `state` field might also be renamed `enumCtx`.